tools/dbTools.py

```diff
@@ -1,50 +1,54 @@
 import sqlite3
-
-
-def check_db():
-    connection = sqlite3.connect('db.sqlite3')
-    cursor = connection.cursor()
-
-    # If articles table does not exist, create it
-    cursor.execute(
-        """
-        CREATE TABLE IF NOT EXISTS article
-        (
-            id INTEGER PRIMARY KEY,
-            title VARCHAR(64),
-            url VARCHAR(255),
-            scrap_again INTEGER(1) DEFAULT(0),
-            Timestamp DATETIME DEFAULT CURRENT_TIMESTAMP
-        );
-        """
-    )
-
-    connection.commit()
-    connection.close()
-
-
-def is_scraped(url):
-    #  Init sqlite db to filter PDFs
-    connection = sqlite3.connect('db.sqlite3')
-    cursor = connection.cursor()
-
-    cursor.execute(
-        "SELECT id FROM article WHERE url LIKE ? AND scrap_again=0",
-        ('%' + url,)
-    )
-    result = cursor.fetchone()
-    connection.close()
-    return result if result else False
-
-
-def insert_article(title, url):
-    #  Init sqlite db to filter PDFs
-    connection = sqlite3.connect('db.sqlite3')
-    cursor = connection.cursor()
-    cursor.execute(
-        "INSERT INTO article (title, url) VALUES (?, ?)",
-        (title, url)
-    )
-
-    connection.commit()
-    connection.close()
+import logging
+
+
+class SQLite3Connector:
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.connection = sqlite3.connect('db.sqlite3')
+        self.cursor = self.connection.cursor()
+        self._check_db()
+
+    def __del__(self):
+        self.connection.commit()
+        self.cursor.close()
+        self.connection.close()
+
+    def _execute(self, query, params=()):
+        try:
+            self.cursor.execute(query, params)
+        except Exception as error:
```

Comment:
Logging here is fine, but I’m surprised you wouldn't raise an exception, which would allow a caller to react to the error – whether that's retry, modify their call, or try again.